Kevin refactor

0spooky2me opened the pull request:
This is a refactored version of the code we had already been using.  It's been updated to follow C++ style conventions and be more coherent.

0spooky2me committed:
Refactored entirety of previous working codebase.

Changes:
Namespaces introduced to encapsulate global variables and detail the exact location of functions.
Redundant code removed.
Existing code optimised for both space and efficiency.
Altered certain functions for increased functionality including
	FollowTape([Alter speed], [bias turning])
Miscellaneous nameing improvements
Miscellaneous style updates

0spooky2me committed:
Fixed formatting and renamed 'main' class from .cpp to .ino

0spooky2me commented:
I've kept away from any OOP design.  There may be a time or a place for it, but it's probably best that it's neither here nor now.

jtroo commented:
Skimmed over it for style, looks pretty good. I assume you know what you're doing since you're the one who knows cpp conventions. Concerns I have is merging with Rachel's additions in her unstable branch. You two will probably have to work together on that to do it manually.

0spooky2me commented:
Right now this is up-to-date with unstable.

0spooky2me committed:
Included a menu to choose testing states and improved strategies

Formalised the strategy picking system
Setup a framework for initialisation of strategies

0spooky2me committed:
Bugfixes: Corrected pivot servo test issues

0spooky2me committed:
fixed clawarmtest bug
forearm value used for both forearm and basearm
